libraries/Microsoft.Bot.Builder/IFrame.cs

```diff
@@ -1,19 +1,18 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace Microsoft.Bot.Builder
 {
     public interface IFrame
     {
-        IFrame Parent { get; set; }
+        IFrame Parent { get; }
 
         void AddSlot(IReadWriteSlot slot);
```

Comment:
I would highly recommend this design be changed. The coupling between this method and the `Slot` constructor is extremely confusing.

I would instead suggest `Slot` actually be removed entirely as a `public` class and this method be changed to a factory method like so: 

```CSharp
IReadWriteSlot<T> AddSlot<T>(SlotDefinition slotDefinition)
```
Internally the implementation then has the responsibility to construct the now `internal` `Slot` class and introduce itself as the `Frame` and users do not have to deal with that coupling at all.